migrate.py

```diff
@@ -1,38 +1,80 @@
 from etleap.api import EtleapApi, EtleapApiException
 
-# US API credentials
-us_etleap_access_key = '...'
-us_etleap_secret_key = '...'
+# env1 API credentials
+# This is where pipelines will be migrated from
 
-# EU API credentials
-eu_etleap_access_key = '...'
-eu_etleap_secret_key = '...'
+env1_base_url = "https://api.etleap.com/api/v2"
+env1_etleap_access_key = '<add here>'
+env1_etleap_secret_key = '<add here>'
+
+# env2 API credentials
+# This is where pipelines will be migrated to
+
+env2_base_url = "https://api.etleap.com/api/v2"
+env2_etleap_access_key = '<add here>'
+env2_etleap_secret_key = '<add here>'
+
+
+# Maps a connection from the env1 to env2
+# Update this as needed in the form of 'env1_connection_id': 'env2_connection_id'
+# If a pipeline has a connection (source or destination) that is not in this map, it will be skipped
 
-# Maps a connection from the US to the EU environment
 connection_map = {
-    'c0Bz8SOK': 'BTIfFpQw',
-    'JSS8ABBJ': 'rOpIIrZl'
+    'env1_connection_id1': 'env2_connection_id1', 
+    'env1_connection_id2': 'env2_connection_id2'
 }
 
-# Only migrated pipelines for sources in this list
-sources_to_migrate = ['c0Bz8SOK']
+# Any pipelines included in this array will attempt to be migrated
+
+pipeline_ids_to_migrate = [
+    # 'pipeline_id_1',
+    # 'pipeline_id_2'
+    ]
 
-us_client = EtleapApi(us_etleap_access_key, us_etleap_secret_key)
-pipelines = us_client.get_pipelines()
+# Any pipelines that ingest from sources in this list will attempt to be migrated
+# Use this if you are doing a bulk migration for all pipelines from this connection
 
-from_source = [ p for p in pipelines if p.source['connectionId'] in sources_to_migrate ]
+sources_to_migrate = [
+    # 'source_connection_id_1', 
+    # 'source_connection_id_2'
+    ]
 
-for p in from_source:
+# -------------------------------------------
+# ------- DO NOT EDIT BELOW THIS LINE -------
+# -------------------------------------------
+
+env1_client = EtleapApi(env1_etleap_access_key, env1_etleap_secret_key, env1_base_url)
+pipelines = env1_client.get_pipelines()
+
+# Only flag pipelines for migration if they match the criteria above 
+from_env1 = [
+    p for p in pipelines
+    if (
+        (sources_to_migrate and p.source['connectionId'] in sources_to_migrate)
+        or (pipeline_ids_to_migrate and p.id in pipeline_ids_to_migrate)
+    )
+]
+
+for p in from_env1:
     if (p.destination['connectionId'] in connection_map.keys()):
```

Comment:
We should probably fail fast if the source or destination connection is not in the mapping. We'll definitely fail when we try to create the pipeline in the second environment.